packages/account-abstraction/src/SmartAccountAPI.ts

```diff
@@ -84,15 +87,18 @@ export class SmartAccountAPI extends BaseAccountAPI {
     readonly index = 0,
     overheads?: Partial<GasOverheads>
   ) {
+    // Note: todo: clientConfig could itself have PaymasterAPI instead of paymasterUrl or dappAPIKey
     super(provider, entryPoint, clientConfig, accountAddress, overheads)
-    if (!clientConfig.dappAPIKey || clientConfig.dappAPIKey === '') {
```

Comment:
this would change to accepting instance of paymaster instead of the url 
and client config would change as per changes in modular_sdk branch
@tomarsachin2271 @talhamalik883 


(currently hack is only token paymaster instance is created by default from the given url)

Reply:
also contemplating if we should have different classes for different paymasters. 
because second arg paymasterServiceData is very generic and serves both paymasters we currently plan to support.

in favour of keeping separate classes in token paymaster class can have it's specific methods implemented which needs to talk to paymaster service to retrieve some information.